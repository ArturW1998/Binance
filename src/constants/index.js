export const ACTION_TYPES = {
  UPDATE_MARKET_PAIRS: 'UPDATE_MARKET_PAIRS',
  SET_ACTIVE_MARKET: 'SET_ACTIVE_MARKET',
  TOGGLE_SOCKET_STREAMS: 'TOGGLE_SOCKET_STREAMS',
};

export const PATHS = {
  BASE_PATH: 'wss://stream.binance.com',
  STREAM_PATH: '/stream',
  STREAM_PARAM: 'streams=',
};

export const LISTS = {
  PAIRS_LIST: ['BNB', 'BTC', 'ETH', 'USDT'],
  DOTS_LIST: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
  TABLE_DATA_LIST: [
    { title: 'Pair' },
    { title: 'Latest Price' },
    { title: 'Open' },
    { title: 'High' },
    { title: 'Low' },
    { title: 'Volume' },
  ],
};

export const CONFIG = {
  BASE_NAME: '/Binance/',
  LOCALSTORAGE_KEY: 'markets',
};
