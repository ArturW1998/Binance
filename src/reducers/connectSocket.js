import { TOGGLE_SOCKET_STREAMS } from '../constants';

const initState = true;

const connectSocket = (state = initState, { type, status }) => {
  switch (type) {
    case TOGGLE_SOCKET_STREAMS:
      return status;
    default:
      return state;
  }
};

export default connectSocket;
