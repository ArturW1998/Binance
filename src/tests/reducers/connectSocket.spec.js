import { ACTION_TYPES } from '../../constants';
import connectSocket, { initState } from '../../reducers/connectSocket';

describe('connectSocket reducer', () => {
  describe('connectSocket reducer with action', () => {
    const { TOGGLE_SOCKET_STREAMS } = ACTION_TYPES;

    const action = {
      type: TOGGLE_SOCKET_STREAMS,
      status: false,
    };

    it(`${TOGGLE_SOCKET_STREAMS}`, () => {
      const { status } = action;

      expect(connectSocket(initState, action)).toEqual(status);
    });

    it(`should handle ${TOGGLE_SOCKET_STREAMS}`, () => {
      expect(connectSocket(initState, action)).toMatchSnapshot();
    });
  });

  describe('connectSocket reducer initial state', () => {
    const action = {};

    it('should return the initial state', () => {
      expect(connectSocket(initState, action)).toEqual(initState);
    });

    it('returns properly', () => {
      expect(connectSocket(initState, action)).toMatchSnapshot();
    });
  });
});
