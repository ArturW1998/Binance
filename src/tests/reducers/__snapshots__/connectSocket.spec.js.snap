// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectSocket reducer initial state returns properly 1`] = `true`;

exports[`connectSocket reducer should handle TOGGLE_SOCKET_STREAMS 1`] = `false`;
